http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2SampleResult.java

```diff
@@ -1,432 +1,427 @@
+   * content-type if necessary.
+   * <p>
+   * Updates the dataEncoding field if the content-type is found.
+   *
+   * @return the dataEncoding value as a String
+   */
+  @Override
+  public String getDataEncodingNoDefault() {
+    if (super.getDataEncodingNoDefault() == null && getContentType()
+        .startsWith("text/html")) { // $NON-NLS-1$
+      byte[] bytes = getResponseData();
+      // get the start of the file
+      String prefix = new String(bytes, 0, Math.min(bytes.length, 2000),
+          Charset.forName(DEFAULT_HTTP_ENCODING));
+      // Preserve original case
+      String matchAgainst = prefix.toLowerCase(java.util.Locale.ENGLISH);
+      // Extract the content-type if present
+      final String METATAG = "<meta http-equiv=\"content-type\" content=\""; // $NON-NLS-1$
+      int tagstart = matchAgainst.indexOf(METATAG);
+      if (tagstart != -1) {
+        tagstart += METATAG.length();
+        int tagend = prefix.indexOf('\"', tagstart); // $NON-NLS-1$
+        if (tagend != -1) {
+          final String ct = prefix.substring(tagstart, tagend);
+          setEncodingAndType(ct);// Update the dataEncoding
+        }
+      }
+    }
+    return super.getDataEncodingNoDefault();
+  }
+
+  /*
+   * (non-Javadoc)
+   *
+   * @see org.apache.jmeter.samplers.SampleResult#getSearchableTokens()
+   */
+  @Override
+  public List<String> getSearchableTokens() throws Exception {
+    List<String> list = new ArrayList<>(super.getSearchableTokens());
+    list.add(getQueryString());
+    list.add(getCookies());
+    return list;
+  }
+
+  public void setPendingResponse(boolean pendingResp) {
+    pendingResponse = pendingResp;
+  }
+
+  public boolean isPendingResponse() {
+    /*boolean ret = pendingResponse;
+    SampleResult[] sons = this.getSubResults();
+    int i = 0;
+    if (sons.length != 0) {
+      while ((i < sons.length) && (!ret)) {
+        HTTP2SampleResult h = (HTTP2SampleResult) sons[i];
+        i++;
+        if (h.isSecondaryRequest()) {
+          ret = ret || (h.isPendingResponse());
+        }
+      }
+    }
+    return ret;*/
+    return pendingResponse;
+  }
+
+  public void setRequestId(String id) {
+    requestId = id;
+  }
+
+  public String getRequestId() {
+    return requestId;
+  }
+
+
+  public boolean isSecondaryRequest() {
+    return secondaryRequest;
+  }
+
+  public void setSecondaryRequest(boolean secondaryRequest) {
+    this.secondaryRequest = secondaryRequest;
+  }
+
+  public void addPendingResult(HTTP2SampleResult pendingResult) {
+    this.pendingResults.add(pendingResult);
+  }
+
+  public JMeterVariables getThreadVars (){
+    return threadVars;
+  }
+
+  public Queue<HTTP2SampleResult> getPendingResults() {
+    return pendingResults;
+  }
+
+  public void notifySample() {
+    HTTP2SampleResult parent = (HTTP2SampleResult) this.getParent();
+    if (parent != null) {
+      parent.notifySample();
+    }
+    if (parent == null && !isPendingResponse()) {

```

Comment:
consider using `else if (!isPendingResponse())` to remove unnecessary check for `parent == null`